addons/mrp/wizard/mrp_product_produce.py

```diff
@@ -119,6 +121,8 @@ def _record_production(self):
                     })
                 line.move_id = move_id.id
 
+        if self.consumption == 'strict':
+            self._check_strict_consumption(self.workorder_line_ids)
```

Comment:
i don't get why this check is done in the middle of everything, i would move it to the first line of record_production

Reply:
now that we get why, add a comment "call _check_strict_consumption" when we are sur every workorder line has an associated move

Reply:
move to the start of update raw move so that the code is shared for the tablet view